packages/daemon/src/app/controller/base.ts

```diff
@@ -0,0 +1,68 @@
+import { Context, inject } from 'midway';
+import { ObjectSchema } from 'joi';
+import * as HttpStatus from 'http-status';
+import ServerSentEmitter from '../../utils/emitter';
+import { LogManager } from '../../service/log-manager';
+
+export class BaseController {
+  @inject()
+  ctx: Context;
+  /**
+   * validata the data by schema
+   * @param schema the schema
+   * @param data the data validated
+   */
+  validate(schema: ObjectSchema, data: any) {
+    const { error } = schema.validate(data);
+    if (error) {
+      this.ctx.throw(error.message, HttpStatus.BAD_REQUEST);
+    }
+  }
+  /**
+   * send successful response, the status will be set to 204 or 200,
+   * depend on if data is undefined
+   * @param data data returned
+   * @param status success code, should be [200, 299], default is 200
+   */
+  success(data?: any, status?: number): void {
+    this.ctx.body = data;
+    if (typeof status !== 'undefined') {
+      this.ctx.status = status;
+    } else {
+      this.ctx.status = typeof data === 'undefined' ? HttpStatus.NO_CONTENT : HttpStatus.OK;
+    }
+  }
+}
```

Comment:
How about extending a `success` method in `this.ctx`(inside `daemon/app/extend/context.ts`)? Then the usage of `this.ctx.success` and `this.ctx.throw` are more consistent.

Type definition could be extending globally, in `/path/to/type/midway.d.ts`:
```ts
declare module 'egg' {
  interface Context {
    success(): void
  }
}
```

Reply:
Nice! I'll update the BaseController.

Reply:
I checked the issues for how to extend context, found this:
https://github.com/koajs/koa/issues/473
For now, I defined the `success` function in `bootstrap.js`. If there is any way  better, I'll work for it.

Reply:
I think I know what's your meaning.
https://eggjs.org/en/tutorials/progressive.html#prototype-of-plugin

Reply:
Seems that I forgot to refer a link about extend context last time😂, sorry for that. You can refer to the following document supported by eggjs
- [Extend EGG](https://eggjs.org/en/basics/extend.html#access-method-2)

daemon/app/extend/context.ts:
```ts
import { Context } from 'midway'

export function success(this: Context, data?: any, status?: number) {
  // do something...
}
```

Reply:
@FeelyChau I noticed that you have extended it in context, but TypeScript can not understand framework conventions, we could extend the type definition globally:

```ts
declare module 'egg' {
  interface Context {
    success(data?: any, status?: number): void
  }
}
```

It looks a bit fragmented, but I think it’s a worthwhile improvement.

Reply:
Done.